src/components/Calendar/CheckinModal.js

```diff
@@ -1,79 +1,121 @@
+    id,
+    eventStatus: checkedIn,
+    end,
+    mentor,
+    resourceId,
+    start,
+    student,
+    title,
+  } = details;
+  const dayString = start ? start.toLocaleDateString() : '';
+  const startString = start ? start.toLocaleTimeString() : '';
+  const endString = end ? end.toLocaleTimeString() : '';
+  
+  const [status, setStatus] = useState(deriveStatus({ start, end, checkedIn }));
+
+  useManageStatus({
+    checkedIn,
+    end,
+    setStatus,
+    start,
+    status,
+  });
+
+  useEffect(() => {
+    setStatus(deriveStatus({ start, end, checkedIn }));
+  }, [details]);
 
   const handleOk = () => {
-    props.setIsModalVisible(false);
+    setIsModalVisible(false);
   };
-
+  
   const handleCancel = () => {
-    props.setIsModalVisible(false);
+    setIsModalVisible(false);
   };
 
-  const day = props.details.start.toLocaleDateString();
-  const start = props.details.start.toLocaleTimeString();
-  const end = props.details.end.toLocaleTimeString();
-
-  const checkedIn = props.details.checkedIn;
+  if (!start) return (<>loading</>);
 
   const handleCheckIn = () => {
-    props.setClickSelected({
-      ...props.details,
-      checkedIn: !props.details.checkedIn,
-    });
-  };
+    const updatedEvent = {
+      ...details,
+      eventStatus: !checkedIn,
+    };
+    setClickSelected(updatedEvent);
+    setEvents((oldEvents) => (
+      oldEvents.map((event) => event.id === id
+        ? updatedEvent
+        : event
+      )
+    ));
+  }
 
   return (
-    <Modal
-      visible={props.isModalVisible}
+    <Modal 
+      visible={isModalVisible}
       onOk={handleOk}
       onCancel={handleCancel}
-      style={{ textAlign: 'center' }}
+      style={{ textAlign: "center"}}
     >
-      <h6
-        style={{
-          textTransform: 'uppercase',
-          fontSize: '.75rem',
-          paddingBottom: '1rem',
-          color: props.details.checkedIn ? '#52c41a' : 'red',
-        }}
-      >
+      <h6 style={{
+          textTransform: 'uppercase', 
+          fontSize: '.75rem', 
+          paddingBottom: '1rem', 
+          color: checkedIn ? '#52c41a' : 'red'
+      }}>
         Status: {checkedIn ? 'checked-in' : 'not checked-in'}
       </h6>
-      <h2>{props.details.title}</h2>
-      {
-        //If the details haven't loaded yet do not display, to handle toLocalDateString() error
-        day && (
-          <p>
-            <strong>When: </strong>
-            {day} {start} - {end}
-          </p>
-        )
-      }
-      <p>
-        <strong>Mentor: </strong>
-        {props.details.mentor}
-      </p>
-      <p>
-        <strong>Student: </strong>
-        {props.details.student}
-      </p>
-      <p>
-        <strong>Computer: </strong>
-        {props.details.resourceId}
-      </p>
-      <Switch
-        checked={checkedIn}
-        checkedChildren="Student Checked In"
-        unCheckedChildren="Check-in Student"
-        onChange={handleCheckIn}
-      />
+      <h6>ExactStatus: {status}</h6>
```

Comment:
I don't know if we need exact status and status in the check-in modal. Maybe we just pick one? The exact status isn't written in friendly though, so if we go with that maybe we remove the underscores.

Reply:
it's debug code; I've removed it.